packages/adapters/ethers/src/client.ts

```diff
@@ -175,889 +95,394 @@ export class EthersAdapter {
-    const providerType = ProviderUtil.state.providerIds['eip155']
-
-    if (providerType === ConstantsUtil.EIP6963_CONNECTOR_ID) {
-      if (currentActiveWallet) {
-        const currentProvider = this.EIP6963Providers.find(
-          provider => provider.info.name === currentActiveWallet
-        )
-
-        if (currentProvider) {
-          this.appKit?.setConnectedWalletInfo({ ...currentProvider.info }, this.chainNamespace)
-        }
-      }
-    } else if (providerType === ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID) {
-      const provider = ProviderUtil.getProvider('eip155')
-
-      if (provider?.session) {
-        this.appKit?.setConnectedWalletInfo(
-          {
-            ...provider.session.peer.metadata,
-            name: provider.session.peer.metadata.name,
-            icon: provider.session.peer.metadata.icons?.[0]
-          },
-          this.chainNamespace
-        )
-      }
-    } else if (providerType === ConstantsUtil.COINBASE_SDK_CONNECTOR_ID) {
-      const connector = this.appKit
-        ?.getConnectors()
-        .find(c => c.id === ConstantsUtil.COINBASE_SDK_CONNECTOR_ID)
-
-      this.appKit?.setConnectedWalletInfo(
-        { name: 'Coinbase Wallet', icon: this.appKit?.getConnectorImage(connector) },
-        this.chainNamespace
-      )
-    } else if (currentActiveWallet) {
-      this.appKit?.setConnectedWalletInfo({ name: currentActiveWallet }, this.chainNamespace)
+  private removeProviderListeners(provider: Provider | CombinedProvider) {
+    if (this.providerHandlers) {
+      provider.removeListener('disconnect', this.providerHandlers.disconnect)
+      provider.removeListener('accountsChanged', this.providerHandlers.accountsChanged)
+      provider.removeListener('chainChanged', this.providerHandlers.chainChanged)
+      this.providerHandlers = null
     }
   }
 
-  private syncRequestedNetworks(caipNetworks: CaipNetwork[]) {
-    const uniqueChainNamespaces = [
-      ...new Set(caipNetworks.map(caipNetwork => caipNetwork.chainNamespace))
-    ]
-    uniqueChainNamespaces.forEach(chainNamespace => {
-      this.appKit?.setRequestedCaipNetworks(
-        caipNetworks.filter(caipNetwork => caipNetwork.chainNamespace === chainNamespace),
-        chainNamespace
-      )
-    })
-  }
-
-  public async switchNetwork(caipNetwork: CaipNetwork) {
-    async function requestSwitchNetwork(provider: Provider) {
+  public async switchNetwork(params: AdapterBlueprint.SwitchNetworkParams): Promise<void> {
+    const { caipNetwork, provider, providerType } = params
+    if (providerType === 'WALLET_CONNECT') {
+      ;(provider as UniversalProvider).setDefaultChain(String(`eip155:${String(caipNetwork.id)}`))
+    } else if (providerType === 'AUTH') {
+      const authProvider = provider as W3mFrameProvider
+      await authProvider.switchNetwork(caipNetwork.id)
+      await authProvider.connect({
+        chainId: caipNetwork.id
+      })
```

Comment:
you can technically directly connect here, using new chain ID!

Reply:
Could you elaborate?